elide-core/src/main/java/com/yahoo/elide/core/security/executors/AggregationStorePermissionExecutor.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2021, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.core.security.executors;
+
+import com.yahoo.elide.annotation.ReadPermission;
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.core.security.ChangeSpec;
+import com.yahoo.elide.core.security.permissions.ExpressionResult;
+import com.yahoo.elide.core.security.permissions.expressions.Expression;
+import com.yahoo.elide.core.type.Type;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.annotation.Annotation;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * Permission Executor for all models managed by aggregation datastore.
+ */
+@Slf4j
+public class AggregationStorePermissionExecutor extends AbstractPermissionExecutor {
+
+    public AggregationStorePermissionExecutor(RequestScope requestScope) {
+        super(log, requestScope);
+    }
+
+    /**
+     * Checks user checks for the requested fields.
+     * expression = (field1Rule OR field2Rule ... OR fieldNRule)
+     * @param annotationClass annotation class
+     * @param resource resource
+     * @param requestedFields the list of requested fields
+     * @param <A>
+     * @return ExpressionResult - result of the above any field expression
+     */
+    @Override
+    public <A extends Annotation> ExpressionResult checkPermission(Class<A> annotationClass,
+                                                                   PersistentResource resource,
+                                                                   Set<String> requestedFields) {
+        if (!annotationClass.equals(ReadPermission.class)) {
+            return ExpressionResult.FAIL;
+        }
+
+        Expression expression = expressionBuilder.buildUserCheckAnyExpression(
```

Comment:
I'm not sure this is what you want here from the Permission Builder.  UserCheckAnyExpression will look at a combination of field and entity for each field.